src/btree/bt_curnext.c

```diff
@@ -759,16 +748,25 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
             WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
         }
         cbt->page_deleted_count = 0;
-skip_page:
+
         if (F_ISSET(cbt, WT_CBT_READ_ONCE))
             LF_SET(WT_READ_WONT_NEED);
-        WT_ERR(__wt_tree_walk(session, &cbt->ref, flags));
+
+        /*
+         * If we are running with snapshot isolation, and considering tombstones, we could
```

Comment:
The flag `WT_CURSTD_IGNORE_TOMBSTONE` is bit misleading. Here we are skipping the pages because we are not interested in returning a tombstones.

Reply:
Fixed.